src/navigator/Navigator.tsx

```diff
@@ -473,6 +474,11 @@ const settingsScreens = (Navigator: typeof Stack) => (
       name={Screens.SignInWithEmail}
       options={noHeader}
       component={SignInWithEmail}
+      initialParams={{
```

Comment:
As far as I'm aware, with react-navigation v6, there is no way to pass a set of "initial params" to the initial route on the navigator itself. This was not the case in earlier versions. Instead, you must provide an `initialParams` object to screens themselves, which, as far as I'm aware, will be used in place of params on the first call if none are present.

Only one screen during onboarding actually has props, the `SignInWithEmail` screen. In my first pass at this, I stored these params in Redux, but since there's no easy way to set these initial props on screens dynamically, I opted for this approach instead. It's kind of lame, having this one special case, but I think it's probably OK.

Another thing to note here, the `SignInWithEmail` screen has a back button, which, if it's the initial route, doesn't do anything. This (and the corresponding changes within the screen itself) cause the back button to be hidden if the screen is being rendered as the initial route. It's not really the end of the world if the back button is there and doesn't do anything, but I thought it would be cleaner to hide it.

Reply:
Is there a way we can check if back is possible? would be nice to avoid the `hideBack` option if so. Also I think @MuckT is doing an upgrade to v7, wonder if v7 has a better way of handling this?

Reply:
There's a [canGoBack](https://reactnavigation.org/docs/6.x/navigation-prop/#cangoback) method in v6, which the documentation states should not be used for rendering purposes since it doesn't trigger a re-render. I think in this case the lack of re-rendering is probably fine, right, since there would _never_ be a way to go back from the initial screen.

Reply:
Looks like [canGoBack](https://reactnavigation.org/docs/navigation-object/#cangoback) is supported in v7.